homeassistant/components/plugwise/binary_sensor.py

```diff
@@ -113,8 +113,8 @@ def __init__(
             self._entity_name = f"Smile {self._entity_name}"
 
     @callback
-    def _async_process_data(self) -> None:
-        """Update the entity."""
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
         if not (data := self.coordinator.data.devices.get(self._dev_id)):
             LOGGER.error("Received no data for device %s", self._dev_id)
             self.async_write_ha_state()
```

Comment:
Should we call super here instead?

Reply:
Yeah probably makes sense, I feel like this part should be killed in another fashion, not sure how yet.

Will adjust it as suggested in a bit.

Reply:
Done